hist/hist/src/TGraph2D.cxx

```diff
@@ -1028,22 +1028,23 @@ TH2D *TGraph2D::GetHistogram(Option_t *option)
       hymin = ymin - fMargin * (ymax - ymin);
       hxmax = xmax + fMargin * (xmax - xmin);
       hymax = ymax + fMargin * (ymax - ymin);
-      if (TMath::Abs(hxmax - hxmin) < 0.0001) {
-         if (TMath::Abs(hxmin) < 0.0001) {
-            hxmin = -0.01;
-            hxmax =  0.01;
+      Double_t epsilon = 1e-9;
```

Comment:
I am not sure epsilon is small enough. If I have some points which are around 10^-10, it will not work either setting limits to 0.01. 
What is done currently in the TGraph ? 
Would not it better to look also at the relative difference and have a check to avoid numerical precision problems happening at the level of a relative epsilon = 10^-16 ?